### PR TITLE
Enable GCD on non-Linux POSIX if not configured

### PR DESCRIPTION
**Pull request by BurningEnlightenment**
I assumed CMake would take care of this, but I was mistaken.

### CODE REVIEW

File: include/llfio/v2.0/detail/impl/dynamic_thread_pool_group.ipp [hunk 1]
```diff
@@ -49,6 +49,10 @@ Distributed under the Boost Software License, Version 1.0.
 #include <condition_variable>
 #include <thread>
 #elif __has_include(<dispatch/dispatch.h>)
+// enable GCD on non-Linux if available but not configured
+#ifndef
```
Comment (ned14) [inline include/llfio/v2.0/detail/impl/dynamic_thread_pool_group.ipp:53]: ?

**Comment by ned14** [comment]
I fixed the quickcpplib constexpr string view failure upstream

**Comment by BurningEnlightenment** [comment]
The MacOS unit test failure doesn't seem to be connected to this PR: 
```
test 4
      Start  4: llfio_sl--clone_extents

4: Test command: /Users/runner/work/llfio/llfio/prebuilt/bin/llfio_sl--clone_extents "--reporter" "junit" "--out" "/Users/runner/work/llfio/llfio/prebuilt/bin/llfio_sl--clone_extents.junit.xml"
4: Working Directory: /Users/runner/work/llfio/llfio/prebuilt
4: Test timeout computed to be: 900
4: 
4: integration/llfio/algorithm/clone_extents : Tests that llfio::file_handle::clone_extents() of partial extents works as expected
4: 
4: 
4: integration/llfio/algorithm/clone_extents:
4: Tests that llfio::file_handle::clone_extents() of partial extents works as expected
4: 
4: Round 1: Cloning 18018052-4710497 to offset 32569981 ...
4: llfio: Freeing large pages failed
4: libc++abi: terminating
 4/93 Test  #4: llfio_sl--clone_extents ...................Subprocess aborted***Exception:   0.29 sec
```

**Comment by ned14** [comment]
Wish I had a Mac to test this stuff :(

Is it spurious or definitely repeatable?

**Comment by BurningEnlightenment** [comment]
> Wish I had a Mac to test this stuff :(
> Is it spurious or definitely repeatable?

I don't own one either. I just copy-pasted the CI log, but I might be able to borrow one from a work colleague tomorrow.

**Comment by BurningEnlightenment** [comment]
> Is it spurious or definitely repeatable?

It occurred again, so it seems repeatable. Sadly I didn't get time on a MacOS computer.

The windows test failure is a byte_socket handle timeout which isn't caused by this AFAICT. So this is ready to be merged or am I missing something?

**Pull request merged**